processing/src/main/java/org/apache/druid/segment/NestedCommonFormatColumnHandler.java

```diff
@@ -100,7 +100,7 @@ public Comparator<ColumnValueSelector> getEncodedValueSelectorComparator()
   {
     if (castTo != null) {
       return (s1, s2) ->
-          castTo.getStrategy().compare(
+          castTo.getNullableStrategy().compare(
```

Comment:
Nit: you can get this once outside of the lambda instead of calling the `get`ter over and over again.